### PR TITLE
Fix infinite loop in bulk translation updater

### PR DESCRIPTION
**Pull request by millerdev**
http://manage.dimagi.com/default.asp?168791

This bug was found during copy/paste QA, but applies to master as well.

cc @sravfeyn

### CODE REVIEW

File: src/javaRosa.js [hunk 1]
```diff
@@ -1117,9 +1117,7 @@ define([
 
         var items = getItextItemsFromMugs(form, true);
         items = _.object(_.map(items, function (item) { return [item.id, item]; }));
-        cells = nextRow();
-        while (cells) {
-            // what's the point of creating items here?
+        for (cells = nextRow(); cells; cells = nextRow()) {
```
Comment (orangejenny) [inline src/javaRosa.js]: curious what the bug is here, old & new code look effectively identical to me

Comment (emord) [inline src/javaRosa.js]: I think it's if it hits https://github.com/dimagi/Vellum/pull/420/files#diff-e5b3eca8407b45b132e0147b8392192cR1124 then it won't increment the row.

g2m when tests are done

Comment (millerdev) [inline src/javaRosa.js]: yeah, the `continue` skipped `cells = nextRow()`, which was previously at the end of the loop.

Comment (czue) [inline src/javaRosa.js]: huh, i did not know that javascript assignment operators returned the value.

Comment (millerdev) [inline src/javaRosa.js]: Interesting comment, @czue. I do not believe I am using that feature here, am I? A more concise way to write this is
```js
while (cells = nextRow()) {
```
but our lint tool doesn't like it (and I think that statement looks pretty misleading), so I opted for the `for` loop instead.

Comment (czue) [inline src/javaRosa.js]: Unless I'm missing something I think you're using the feature. The statement is an assignment, which in python would evaluate to nothing, but in javascript evaluates to the right side of the assignment. Therefore I think both constructs rely on that feature.

Comment (millerdev) [inline src/javaRosa.js]: Not understanding... This is now I understand it:
```
for (<assignment, result ignored>; <test, result used>; <increment, result ignored>) {
```
What am I missing?

Comment (czue) [inline src/javaRosa.js]: sorry you're right. i was confused about the ordering

**Pull request merged**